design/security.go

```diff
@@ -0,0 +1,76 @@
+package design
+
+// SecurityMethodKind is a type of security method, according to the
+// swagger specs.
+type SecurityMethodKind int
+
+const (
+	// OAuth2SecurityKind means "oauth2" security type.
+	OAuth2SecurityKind SecurityMethodKind = iota + 1
+	// BasicAuthSecurityKind means "basic" security type.
+	BasicAuthSecurityKind
+	// APIKeySecurityKind means "apiKey" security type.
+	APIKeySecurityKind
+	// OtherSecurityKind is for custom security types, when you want to above
+	// and beyond Swagger.
+	OtherSecurityKind
+)
+
+// SecurityDefinition defines security requirements for an Action
+type SecurityDefinition struct {
+	// NoSecurity is true when you don't want to inherit the resource
+	// (or API-level) security requirements.
+	NoSecurity bool
```

Comment:
Do we need an explicit bool field for this? couldn't that just be another security kind?